### PR TITLE
fix(route): 巴哈姆特電玩資訊站-GNN 新聞

### LINKED ISSUE #13792: 巴哈姆特路由錯誤
### 路由地址

```routes
/gamer/gnn/:category?
```

### 完整路由地址

```fullroutes
/gamer/gnn/1
```

### 相关文档

https://docs.rsshub.app/zh/routes/game#ba-ha-mu-te-gnn-xin-wen

### 预期是什么？

返回正确内容

### 实际发生了什么？

Route requested: /gnn/1
Error message: Cannot read properties of undefined (reading 'replace')
Helpful Information to provide when opening issue:
Path: /gnn/1
Node version: v18.18.2
Git Hash: f4f14bc

### 部署

自架(https://rss-hub-theta.vercel.app/)

### 部署相关信息

_No response_

### 额外信息

```shell
Route requested: /gnn/1
Error message: Cannot read properties of undefined (reading 'replace')
Helpful Information to provide when opening issue:
Path: /gnn/1
Node version: v18.18.2
Git Hash: f4f14bc
```

### 这不是重复的 issue

- [X] 我已经搜索了 [现有 issue](https://github.com/DIYgod/RSSHub/issues)，以确保该错误尚未被报告。

### PR DESCRIPTION
**Pull request by ladeng07**
## Involved Issue / 该 PR 相关 Issue

Close #13792 

## Example for the Proposed Route(s) / 路由地址示例

```routes
/gamer/gnn/1
```

## New RSS Route Checklist / 新 RSS 路由检查表
  
- [ ] New Route / 新的路由
  - [ ] Follows [Script Standard](https://docs.rsshub.app/joinus/advanced/script-standard) / 跟随 [路由规范](https://docs.rsshub.app/zh/joinus/advanced/script-standard)
- [ ] Anti-bot or rate limit / 反爬/频率限制
  - [ ] If yes, do your code reflect this sign? / 如果有, 是否有对应的措施?
- [ ] [Date and time](https://docs.rsshub.app/joinus/advanced/pub-date) / [日期和时间](https://docs.rsshub.app/zh/joinus/advanced/pub-date)
  - [ ] Parsed / 可以解析
  - [ ] Correct time zone / 时区正确
- [ ] New package added / 添加了新的包
- [ ] `Puppeteer`

### CODE REVIEW

**Review by TonyRL — COMMENTED** [review]
https://github.com/DIYgod/RSSHub/blob/fcc077b544b9116dbf67f066bacc5d36f463e8d2/lib/routes/gamer/gnn-index.ts#L95
Consider limiting the number of item in `list` to 50(half of the original) using the common query parameter `limit`.

**Comment by ladeng07** [comment]
> https://github.com/DIYgod/RSSHub/blob/fcc077b544b9116dbf67f066bacc5d36f463e8d2/lib/routes/gamer/gnn-index.ts#L95
> 
> 
> Consider limiting the number of item in `list` to 50(half of the original) with the common query parameter `limit`.

Okay, I've finished this

File: lib/routes/gamer/gnn-index.ts [hunk 1]
```diff
@@ -137,7 +138,7 @@ async function handler(ctx) {
                         component = _$('div.text-paragraph').html();
                     }
                 }
-                item.pubDate = timezone(parseDate(dateStr, 'YYYY-MM-DD HH:mm:ss'), +8);
+                item.pubDate = timezone(parseDate(dateStr, 'YYYY-MM-DD HH:mm:ss'), 0);
```
Comment (TonyRL) [inline lib/routes/gamer/gnn-index.ts:141]: I don't think the date from articles are from UTC +0.

Comment (ladeng07) [reply lib/routes/gamer/gnn-index.ts:141]: Sorry, I got confused about the time zone.

**Pull request merged**